### PR TITLE
Fixing python 3.4 compatibility support. issue #357

### PR DESCRIPTION
**Pull request by singer**
Possible issue with circular imports (https://docs.python.org/3/whatsnew/3.5.html#other-language-changes) which is handled ok it 3.5+ and is giving an error in 3.4

### CODE REVIEW

**Comment by Lonami** [comment]
May fix #357?

**Comment by singer** [comment]
yes it fixes #359

File: telethon_generator/tl_generator.py [hunk 1]
```diff
@@ -117,8 +117,18 @@ def generate_tlobjects(self, scheme_file, import_depth):
...
         os.makedirs(out_dir, exist_ok=True)
+
+        # Generate __init__.py with  relative imports to the namespaces
+        with open(os.path.join(out_dir, '__init__.py'), 'w', encoding='utf-8') as f, \
+                SourceBuilder(f) as builder:
+            builder.writeln(AUTO_GEN_NOTICE)
+            builder.writeln('from . import {}'.format(', '.join(
+                x for x in namespace_tlobjects.keys() if x
+            )))
+            builder.writeln('from .base import *')
```
Comment (singer) [inline telethon_generator/tl_generator.py]: I think here` import *` is used as intended

**Comment by Lonami** [comment]
After some more investigation we found that https://github.com/LonamiWebs/Telethon/commit/ee01724cdb7027c1e38625d31446ba1ea7bade92 is likely the cause. If you would like to test the change I made on https://github.com/LonamiWebs/Telethon/commit/5f0faee0132fe940600bf51356e30c2d73883cca, that would be nice :)

**Comment by singer** [comment]
Nope only this change does not help. 
```
Traceback (most recent call last):
  File "example.py", line 1, in <module>
    from telethon import TelegramClient
  File "/home/singer/envs/telemaster34/Telethon/telethon/__init__.py", line 1, in <module>
    from .telegram_bare_client import TelegramBareClient
  File "/home/singer/envs/telemaster34/Telethon/telethon/telegram_bare_client.py", line 13, in <module>
    from .crypto import rsa, CdnDecrypter
  File "/home/singer/envs/telemaster34/Telethon/telethon/crypto/__init__.py", line 3, in <module>
    from .auth_key import AuthKey
  File "/home/singer/envs/telemaster34/Telethon/telethon/crypto/auth_key.py", line 5, in <module>
    from ..extensions import BinaryReader
  File "/home/singer/envs/telemaster34/Telethon/telethon/extensions/__init__.py", line 6, in <module>
    from .binary_reader import BinaryReader
  File "/home/singer/envs/telemaster34/Telethon/telethon/extensions/binary_reader.py", line 7, in <module>
    from ..tl.all_tlobjects import tlobjects
  File "/home/singer/envs/telemaster34/Telethon/telethon/tl/__init__.py", line 2, in <module>
    from .session import Session
  File "/home/singer/envs/telemaster34/Telethon/telethon/tl/session.py", line 9, in <module>
    from .entity_database import EntityDatabase
  File "/home/singer/envs/telemaster34/Telethon/telethon/tl/entity_database.py", line 5, in <module>
    from ..tl.types import (
  File "/home/singer/envs/telemaster34/Telethon/telethon/tl/types/__init__.py", line 3, in <module>
    from . import phone, storage, channels, contacts, account, auth, updates, payments, help, upload, photos, messages
  File "/home/singer/envs/telemaster34/Telethon/telethon/tl/types/phone.py", line 3, in <module>
    from ...tl import types
ImportError: cannot import name 'types'
```

**Comment by Lonami** [comment]
Closed by https://github.com/LonamiWebs/Telethon/commit/b3ca68b7d9cc7a100a9ba4550c0de442146865f4, sorry I couldn't merge but I appreciate your help!

**Pull request closed without merging**